autopilot/autopilot.go

```diff
@@ -698,3 +729,166 @@ func (ap *Autopilot) hostsHandlerPOST(jc jape.Context) {
 	}
 	jc.Encode(hosts)
 }
+
+func countUsableHosts(cfg api.AutopilotConfig, cs api.ConsensusState, fee types.Currency, currentPeriod uint64, rs api.RedundancySettings, gs api.GougingSettings, hosts []hostdb.Host) (usables uint64) {
+	gc := worker.NewGougingChecker(gs, cs, fee, currentPeriod, cfg.Contracts.RenewWindow)
+	for _, host := range hosts {
+		usable, _ := isUsableHost(cfg, rs, gc, host, 0, 0)
+		if usable {
+			usables++
+		}
+	}
+	return
+}
+
+// evaluateConfig evaluates the given configuration and if the gouging settings
+// are too strict for the number of contracts required by 'cfg', it will provide
+// a recommendation on how to loosen it.
+func evaluateConfig(cfg api.AutopilotConfig, cs api.ConsensusState, fee types.Currency, currentPeriod uint64, rs api.RedundancySettings, gs api.GougingSettings, hosts []hostdb.Host) (resp api.ConfigEvaluationResponse) {
+	gc := worker.NewGougingChecker(gs, cs, fee, currentPeriod, cfg.Contracts.RenewWindow)
+	for _, host := range hosts {
+		usable, usableBreakdown := isUsableHost(cfg, rs, gc, host, 0, 0)
+		if usable {
+			resp.Usable++
+			continue
+		}
+		resp.Total++
```

Comment:
This will change if you adopt the new struct but it's kind of odd `Total` is not incremented for `usable` hosts.